src/audio/up_down_mixer/up_down_mixer.c

```diff
@@ -320,232 +320,136 @@ static int init_mix(struct comp_dev *dev,
+		ret = -EINVAL;
+		break;
 	}
 
-	if (!cd->mix_routine || ret < 0) {
-		up_down_mixer_free(dev);
-		comp_err(dev, "init_up_down_mixer(): mix routine uninitialized.");
-		return -EINVAL;
+	if (ret < 0) {
+		comp_err(dev, "up_down_mixer_init(): failed to initialize up_down_mix");
+		goto err;
 	}
 
 	return 0;
-}
-
-static struct comp_dev *up_down_mixer_new(const struct comp_driver *drv,
-					  const struct comp_ipc_config *config,
-					  const void *spec)
-{
-	struct comp_dev *dev;
-	int ret;
-
-	comp_cl_info(&comp_up_down_mixer, "up_down_mixer_new()");
 
-	dev = comp_alloc(drv, sizeof(*dev));
-	if (!dev)
-		return NULL;
-
-	ret = init_up_down_mixer(dev, config, spec);
-	if (ret < 0) {
-		comp_free(dev);
-		return NULL;
-	}
-	dev->state = COMP_STATE_READY;
-
-	return dev;
+err:
+	up_down_mixer_free(mod);
+	return ret;
 }
 
-static int up_down_mixer_prepare(struct comp_dev *dev)
+/* just stubs for now. Remove these after making these ops optional in the module adapter */
+static int up_down_mixer_prepare(struct processing_module *mod)
 {
-	int ret;
-
-	if (dev->state == COMP_STATE_ACTIVE) {
-		comp_info(dev, "up_down_mixer_prepare(): Component is in active state.");
-		return 0;
-	}
-
-	ret = comp_set_state(dev, COMP_TRIGGER_PREPARE);
-	if (ret < 0)
-		return ret;
-
-	if (ret == COMP_STATUS_STATE_ALREADY_SET)
-		return PPL_STATUS_PATH_STOP;
-
 	return 0;
 }
 
-static int up_down_mixer_reset(struct comp_dev *dev)
+static int up_down_mixer_reset(struct processing_module *mod)
 {
-	comp_set_state(dev, COMP_TRIGGER_RESET);
 	return 0;
 }
 
-static int up_down_mixer_copy(struct comp_dev *dev)
+static int
+up_down_mixer_process(struct processing_module *mod,
+		      struct input_stream_buffer *input_buffers, int num_input_buffers,
+		      struct output_stream_buffer *output_buffers, int num_output_buffers)
 {
-	struct comp_buffer *source, *sink;
-	struct comp_buffer __sparse_cache *source_c, *sink_c;
+	struct up_down_mixer_data *cd = module_get_private_data(mod);
+	struct comp_dev *dev = mod->dev;
 	uint32_t source_bytes, sink_bytes;
 	uint32_t mix_frames;
 
-	struct up_down_mixer_data *cd = comp_get_drvdata(dev);
-
-	comp_dbg(dev, "up_down_mixer_copy()");
-
-	source = list_first_item(&dev->bsource_list, struct comp_buffer,
-				 sink_list);
-	sink = list_first_item(&dev->bsink_list, struct comp_buffer,
-			       source_list);
-
-	source_c = buffer_acquire(source);
-	sink_c = buffer_acquire(sink);
+	comp_dbg(dev, "up_down_mixer_process()");
 
-	mix_frames = MIN(audio_stream_get_avail_frames(&source_c->stream),
-			 audio_stream_get_free_frames(&sink_c->stream));
+	mix_frames = mod->input_buffers[0].size;
 
-	source_bytes = mix_frames * audio_stream_frame_bytes(&source_c->stream);
-	sink_bytes = mix_frames * audio_stream_frame_bytes(&sink_c->stream);
+	source_bytes = mix_frames * audio_stream_frame_bytes(mod->input_buffers[0].data);
+	sink_bytes = mix_frames * audio_stream_frame_bytes(mod->output_buffers[0].data);
```

Comment:
aren't `input_buffer` / `output_buffer` function parameters already pointing to these arrays?
```
	ret = module_process(mod, mod->input_buffers, num_input_buffers,
			     mod->output_buffers, mod->num_output_buffers);
```

Reply:
@ranj063 ? This seems a generic issue? Why pass input and output buffers to process() if they are already set up to "struct processing_module" in the prepare step. It seems this is a module adapter specific thing and modules themselves should use the input/outputs passed to process instead.

Reply:
@kv2019i yes this is something that needs to cleand up in all modules. Will follow up in another PR

Reply:
I can do this, I created one jira to track this: https://jira.devtools.intel.com/browse/SOFC2-311

only one question, shall we do it right now, or wait for all component move to moduler adapter?

Reply:
@btian1 we can start doing it now.

Reply:
ok, let me submit a PR for this topic.